openhands/server/routes/git.py

```diff
@@ -229,3 +239,268 @@ async def get_repository_branches(
+                # Get the actual creation time from Git
+                agent_file_path = Path(r_agent.source)
+                created_at = _get_file_creation_time(repo_dir, agent_file_path)
+
+                microagents.append(
+                    MicroagentResponse(
+                        name=name,
+                        type='repo',
+                        content=r_agent.content,
+                        triggers=[],
+                        inputs=r_agent.metadata.inputs,
+                        tools=(
+                            [
+                                server.name
+                                for server in r_agent.metadata.mcp_tools.stdio_servers
+                            ]
+                            if r_agent.metadata.mcp_tools
+                            else []
+                        ),
+                        created_at=created_at,
+                        git_provider=git_provider,
+                    )
+                )
+
+            # Add knowledge microagents
+            for name, k_agent in knowledge_agents.items():
+                # Get the actual creation time from Git
+                agent_file_path = Path(k_agent.source)
+                created_at = _get_file_creation_time(repo_dir, agent_file_path)
+
+                microagents.append(
+                    MicroagentResponse(
+                        name=name,
+                        type='knowledge',
+                        content=k_agent.content,
+                        triggers=k_agent.triggers,
+                        inputs=k_agent.metadata.inputs,
+                        tools=(
+                            [
+                                server.name
+                                for server in k_agent.metadata.mcp_tools.stdio_servers
+                            ]
+                            if k_agent.metadata.mcp_tools
+                            else []
+                        ),
+                        created_at=created_at,
+                        git_provider=git_provider,
+                    )
+                )
+
+            logger.info(f'Found {len(microagents)} microagents in {repository_name}')
+            return microagents
+
+        finally:
+            # Clean up temporary directory
+            shutil.rmtree(temp_dir, ignore_errors=True)
+
+    except Exception as e:
+        logger.error(
+            f'Error scanning repository {repository_name}: {str(e)}', exc_info=True
+        )
+        return JSONResponse(
+            content=f'Error scanning repository: {str(e)}',
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+        )
+
+
+def _get_default_domain(git_provider):
+    """Get the default domain for a git provider."""
+    from openhands.integrations.service_types import ProviderType
+
+    domain_map = {
+        ProviderType.GITHUB: 'github.com',
+        ProviderType.GITLAB: 'gitlab.com',
+        ProviderType.BITBUCKET: 'bitbucket.org',
+    }
```

Comment:
NIT: this map also seems to be present in multiple parts of the codebase. Maybe we can have these maps and fetching logic be part of the `ProviderHandler` class